### PR TITLE
Remove old broker e2e test directory

### PR DESCRIPTION
**Pull request by creydr**
## Proposed Changes

- :broom: Removes the empty e2e_broker directory as all broker e2e tests were migrated to the rekt based and moved to `e2e_new`

/hold until #3003 merges

### CODE REVIEW

File: test/e2e-tests.sh [hunk 1]
```diff
@@ -34,15 +34,7 @@ else
   echo "BROKER_CLASS is set to '${BROKER_CLASS}'. Running tests for that broker class."
 fi
 
-if [ "${BROKER_CLASS}" == "KafkaNamespaced" ]; then
-  # if flag exists, only test tests that are relevant to namespaced KafkaBroker
-  echo "BROKER_CLASS is set to 'KafkaNamespaced'. Only running the relevant tests."
-  go_test_e2e -timeout=1h ./test/e2e_broker/... || fail_test "E2E suite failed (directory: ./e2e_broker/...)"
-  success
-fi
```
Comment (pierDipi) [inline test/e2e-tests.sh:42]: This means we can remove the separate `integration-tests-namespaced-broker_eventing-kafka-broker_main` job

Comment (pierDipi) [reply test/e2e-tests.sh:42]: Test infra config is at https://github.com/knative/test-infra/blob/main/prow/jobs_config/knative-sandbox/eventing-kafka-broker.yaml

Comment (creydr) [reply test/e2e-tests.sh:42]: As I understood yes. This should be now covered by the `reconciler-tests-namespaced-broker` job.
@aliok: any protest?

Comment (creydr) [reply test/e2e-tests.sh:42]: PR: https://github.com/knative/test-infra/pull/3758

**Review by pierDipi — APPROVED** [review]
:tada: 

/lgtm
/approve

**Pull request merged**